editor/import/resource_importer_texture.cpp

```diff
@@ -580,48 +711,12 @@ Error ResourceImporterTexture::import(ResourceUID::ID p_source_id, const String
 
 		// Invert the green channel of the image to flip the normal map it contains.
 		if (normal_map_invert_y) {
-			// Inverting the green channel can be used to flip a normal map's direction.
-			// There's no standard when it comes to normal map Y direction, so this is
-			// sometimes needed when using a normal map exported from another program.
-			// See <http://wiki.polycount.com/wiki/Normal_Map_Technical_Details#Common_Swizzle_Coordinates>.
-			const int height = target_image->get_height();
-			const int width = target_image->get_width();
-
-			for (int i = 0; i < width; i++) {
-				for (int j = 0; j < height; j++) {
-					const Color color = target_image->get_pixel(i, j);
-					target_image->set_pixel(i, j, Color(color.r, 1 - color.g, color.b, color.a));
-				}
-			}
+			_invert_y_channel(target_image);
```

Comment:
The `normal_map_invert_y` flag could theoretically be marked as deprecated in the docs, since it can now be accomplished by setting the green channel remap set to inverted green.
It's unclear from user perspective whether `normal_map_invert_y` happens before or after channel remapping, so ideally user should only use one or the other but not both at once. So deprecation warning could help nudging user towards this.

Removing the property seems more difficult due to versioning and compatibility, so that might have to wait for 5.0...

What do you think?

Reply:
Indeed, I think it makes sense to deprecate it and point to the channel remapping options.